src/server/services/crypto.js

```diff
@@ -9,3 +11,33 @@ export async function hashPassword(password) {
 export async function comparePasswords(password, hash) {
   return await bcrypt.compare(password, hash);
 }
+
+export function hashSecret(str) {
+  // We generate a keyed hash for all data instances
+  // to identify them on the blockchain without leaking
+  // any data / to keep them private.
+  const { signature } = web3.eth.accounts.sign(
+    str,
+    process.env.HASH_PRIVATE_KEY,
+  );
+
+  return signature;
+}
+
+export function compareHashSecret(str, hash) {
+  return hashSecret(str) === hash;
+}
+
+export function validateHashSecret(str, hash, address) {
+  const recoveredAddress = web3.eth.accounts.recover(str, hash);
```

Comment:
so, strictly speaking, what's called a hash here isn't actually a hash, it's a signature, and one of the fields inside the web3 signature object is actually the hash (msgHash). 

Pretty sure we can still use signatures the way we talked about using salted hashes/keyed hashes but I'd suggest calling them signatures so we don't get confused about other places where we actually have hashes